test/test_tensor_creation_ops.py

```diff
@@ -1960,6 +1960,99 @@ def test_inference(default_dtype):
         test_inference(torch.float64)
         test_inference(torch.float32)
 
+    @suppress_warnings
+    @onlyCPU
+    def test_tensor_factory_ndarrays(self, device):
+        """
+        Tests torch.tensor for multiple numpy arrays that are embedded (recursively) within list(s). 
+        """
+
+        repeats = 10
+        np_types = list(torch_to_numpy_dtype_dict.values())
+
+        # Simple tests
+        for dtype in np_types:
+            for _ in range(repeats):
+                stack = []
+                stack.append([np.array(1).astype(dtype)])
```

Comment:
Instead of using repeats and calling the nested datastructure a "stack," @ngimel and I suggest handwriting the test cases in a tuple:

```
test_cases = (
  # Case 0: single c-contiguous NumPy array
  (...),
  # Case 1: single f-contiguous NumPy array
  (...),
  ...
)
```

Then each test case can be iterated over at once. The use of "repeats" seems unnecessary. It would also be helpful to add a comment for each test case. That will make it easier to understand test coverage. If a test case is especially complicated then consider using a helper function to generate it.

In addition to test cases for the functionality added in this PR, let's be extra safe and add some test cases for paths not improved by this PR to ensure they're not regressed. For example, a valid input of NumPy arrays nested as ((a, b), c). 

Once the valid inputs are working, for each error message create a "negative" test cases and use assertRaisesRegex to validate it's thrown when appropriate, too.

Reply:
Without including the optimization, ((a,b), c) throws a value error by the original slow path. I have added the negative test cases. Most of the slow path test cases are actually already there and so I added 1 negative test case for the slow path.

I think the way the test cases are laid out currently is more flexible than using a single test_cases (as shown above) without losing readability. For test cases which seem a little vague, I have added comments. I have removed ```repeats```